src/sources/datadog_agent/metrics.rs

```diff
@@ -378,9 +378,13 @@ fn decode_datadog_series_v1(
 
 fn into_metric_tags(tags: Vec<String>) -> MetricTags {
     tags.iter()
-        .map(|tag| {
-            let kv = tag.split_once(':').unwrap_or((tag, ""));
-            (kv.0.trim().to_string(), kv.1.trim().to_string())
+        .map(|tag| match tag.find(':') {
+            Some(colon) => {
+                let key = tag[..colon].to_string();
+                let value: TagValue = (tag[colon + 1..]).to_string().into();
+                (key, value)
+            }
+            None => (tag.to_string(), TagValue::Bare),
         })
         .collect()
 }
```

Comment:
This strikes me as the same as the `statsd` PR, unsurprising given the statsd/dogstatsd relationship - this live in a shared location?

Reply:
Yeah that's a good idea. since they are independent PRs both children of master, I'll wait for the statsd PR to go in, sync it here and update this PR to a shared locale.